firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateAppClient.java

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import static com.google.firebase.appdistribution.FirebaseAppDistributionException.Status.UPDATE_NOT_AVAILABLE;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.CancellationTokenSource;
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.appdistribution.internal.AppDistributionReleaseInternal;
+
+/** Client class for updateApp functionality in {@link FirebaseAppDistribution}. */
+public class UpdateAppClient {
+
+  private TaskCompletionSource<UpdateState> updateAppTaskCompletionSource = null;
+  private CancellationTokenSource updateAppCancellationSource;
+  private UpdateTaskImpl updateTask;
+  private FirebaseApp firebaseApp;
+
+  public UpdateAppClient(@NonNull FirebaseApp firebaseApp) {
+    this.firebaseApp = firebaseApp;
+  }
+
+  @NonNull
+  public UpdateTask getUpdateTask(
+      @NonNull AppDistributionReleaseInternal latestRelease, @NonNull Activity currentActivity)
+      throws FirebaseAppDistributionException {
+
+    if (latestRelease == null) {
```

Comment:
Should this check be in FirebaseAppDistribution? I think if we have the @NonNull annotation on the latestRelease parameter, it might make sense to verify that the cachedLatestRelease is nonnull before calling this method

Reply:
Yep makes more sense outside given the annotation